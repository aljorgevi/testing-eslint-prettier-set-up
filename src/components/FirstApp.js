import PropTypes from "prop-types";

export default function FirstApp({ title, subTitle, name }) {
	return (
		<>
			<h1 data-testid="test-title">{title}</h1>
			<p>{subTitle}</p>
			<p>{name}</p>
		</>
	);
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
	name: "Jorge Mena",
	subTitle: "there is no subTitle",
	// title: 'No hay t√≠tulo',
};
