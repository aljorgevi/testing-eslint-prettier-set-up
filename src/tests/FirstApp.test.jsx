import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import FirstApp from "../components/FirstApp";

describe("Test of <FirstApp />", () => {
	test.skip("must match eith snapshot", () => {
		const title = "Hello World";
		const { container } = render(<FirstApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test("must show the title in h1", () => {
		const title = "Hello World";
		// eslint-disable-next-line no-unused-vars
		const { container, getByText, getByTestId } = render(<FirstApp title={title} />);
		expect(getByText(title)).toBeInTheDocument();
		expect(getByTestId("test-title")).toHaveTextContent(title);

		// const h1 = container.querySelector("h1");
		// expect(h1.innerHTML).toBe(title);
	});

	test("should show the subtitle send by props", () => {
		const subtitle = "Hello World";
		const { getByText } = render(<FirstApp subTitle={subtitle} />);
		expect(getByText(subtitle)).toBeInTheDocument();
	});
});
