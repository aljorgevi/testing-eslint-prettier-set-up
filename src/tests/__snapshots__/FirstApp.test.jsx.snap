// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <FirstApp /> must match eith snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <p>
    there is no subTitle
  </p>
  <p>
    Jorge Mena
  </p>
</div>
`;
